src/main/ApplicationImpl.cpp

```diff
@@ -362,6 +364,24 @@ ApplicationImpl::reportInfo()
     LOG_INFO(DEFAULT_LOG, "info -> {}", getJsonInfo().toStyledString());
 }
 
+std::shared_ptr<BasicWork>
+ApplicationImpl::scheduleSelfCheck()
+{
+    auto& ws = getWorkScheduler();
+    auto& lm = getLedgerManager();
+    auto& bm = getBucketManager();
+    auto& ham = getHistoryArchiveManager();
+
+    LedgerHeaderHistoryEntry lhhe = lm.getLastClosedLedgerHeader();
+
+    std::vector<std::shared_ptr<BasicWork>> seq;
+    seq.emplace_back(bm.getVerifyReferencedBucketsWork());
```

Comment:
I am not sure how viable the `getVerifyReferencedBucketsWork` is in the "async" case (ie when run on startup when the node is "just running" or triggered by the "self-check" endpoint):
* it includes buckets referenced by snapshots not published yet, which can be arbitrary large
* when the node is running, the list of buckets captured at this time is going to change:
  * buckets referenced by the bucket list are going to get deleted on ledger close
  * publish is going to cause some buckets to be deleted as well

Reply:
- Buckets are immutable and refcounted, so these will just be held on to while verifying.
  - It sounds like we want to move this out of the onine phase only to the offline phase anyways, so it won't matter.

Reply:
(done this -- buckets no longer checked online)